### PR TITLE
Add WebFinger Response Objects

### PR DESCRIPTION
**Pull request by nkramer44**
Signed-off-by: nkramer44 <noah.ph.kramer@gmail.com>

## High Level Overview of Change
Add two immutable class definitions representing a JSON Resource Descriptor (JRD) for WebFinger requests.

### Context of Change
PayID dereferencing will now be accomplished using the WebFinger standard as defined [here](https://tools.ietf.org/html/rfc7033).  More details about this flow can be found in the [PayID Discovery RFC](https://github.com/xpring-eng/rfcs/blob/master/payid/src/spec/payid-discovery.md).

We will need these classes to deserialize JRDs to dereference PayIDs.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After
New JRD classes.

## Test Plan
Two JSON serialization/deserialization tests.

Use these to actually dereference a PayID

### CODE REVIEW

File: src/main/java/io/xpring/payid/WebFingerLink.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   * @return A {@link String} representing the target URI.
+   */
+  String href();
+
+  /**
+   * The media type of what the result of dereferencing the link should be.
+   *
+   * @return A {@link String} representing the media type.
+   */
+  String type();
```
Comment (sappenin) [inline src/main/java/io/xpring/payid/WebFingerLink.java:45]: This can be removed.

File: src/main/java/io/xpring/payid/WebFingerLink.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+   * @return A {@link String} representing the relation type.
+   */
+  String rel();
+
+  /**
+   * The target URI of a {@link WebFingerLink}.
+   *
+   * @return A {@link String} representing the target URI.
+   */
+  String href();
```
Comment (sappenin) [inline src/main/java/io/xpring/payid/WebFingerLink.java:38]:
```suggestion
  Optional<String> href();
  Optional<String> template();
```

File: src/main/java/io/xpring/payid/WebFingerJrd.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+import java.util.List;
+
+/**
+ * Represents a JSON Resource Descriptor (JRD) returned from a WebFinger request as defined in RFC-7033.
+ *
+ * <p>
+ * For the purposes of PayID dereferencing, only subject and links are necessary.  Thus, the aliases and properties
```
Comment (sappenin) [inline src/main/java/io/xpring/payid/WebFingerJrd.java:13]:
```suggestion
 * For the purposes of PayID Discovery, only subject and links are necessary.  Thus, the aliases and properties
```

**Comment by nkramer44** [comment]
Closing for now, since this seems to be out of scope for May 7th.

**Pull request closed without merging**